command/agent_test.go

```diff
@@ -2082,3 +2082,118 @@ listener "tcp" {
 		"Points",
 	})
 }
+
+func TestAgent_Quit(t *testing.T) {
```

Comment:
nit: it looks like we are testing two modes within the same test. I would consider perhaps making this test table driven?

Reply:
I actually explicitly want to test that the per-listener config is working properly, i.e. that it's not globally enabling  the API

Reply:
Sounds good!